blockly-core/core/ui/contract_editor/contract_editor.js

```diff
@@ -141,22 +142,31 @@ Blockly.ContractEditor.prototype.create_ = function() {
   );
 
   this.hiddenExampleBlocks_ = [];
+  /** @type {Blockly.SvgTextButton} */
+  this.addExampleButton = new Blockly.SvgTextButton(
+    canvasToDrawOn,
+    "Add Example", // TODO(bjordan): i18n
+    this.addNewExampleBlock_.bind(this)
+  );
+
   this.examplesSectionView_ = new Blockly.ContractEditorSectionView(
     canvasToDrawOn, {
       headerText: "2. Examples", // TODO(bjordan): i18n
       placeContentCallback: goog.bind(function (currentY) {
-        if (this.exampleBlocks.length === 0) {
-          return currentY;
-        }
         var newY = currentY;
         newY += EXAMPLE_BLOCK_SECTION_MAGIN_ABOVE;
-        this.exampleBlocks.forEach(function(block, i) {
-          if (i !== 0) {
-            newY += EXAMPLE_BLOCK_MARGIN_BELOW;
-          }
-          block.moveTo(EXAMPLE_BLOCK_MARGIN_LEFT, newY);
-          newY += block.getHeightWidth().height;
-        }, this);
+        if (this.exampleBlocks.length !== 0) {
```

Comment:
if length is 0, won't the forEach be a noop? should we just have the newY inside the .length check?